### PR TITLE
50 add path in awt and fix bugs in xml

### PR DESCRIPTION
**Pull request by starsasumi**
Still for issue #50. For the arc path type, it's too complicated to add. But I think it's better to merge the rest of paths for now.

Could we just remove arc in LineType?

### CODE REVIEW

**Comment by anqitong** [comment]
The test works fine for me : only one thing : the text is bold in SVG file but not in the window

**Comment by starsasumi** [comment]
@anqitong I noticed this and I would like to discuss it with you guys. In AWT, we can no assign a stroke width to text, but in xml, we can. So we could either keep it this way or disable the width of text in XML. What would be better?

**Comment by anqitong** [comment]
@starsasumi  : problem in the test for #67  : 
- in the window the polyline is not filled in black
- in the svgAndAwt.svg file the polyline is filled in black!

**Comment by anqitong** [comment]
But can you in AWT make a text bold?

**Comment by anqitong** [comment]
I thinkk it is better to disable the width and to choose a default width for pencils

**Comment by starsasumi** [comment]
@anqitong It's strange ... the polyline is not filled in my computer (I fixed this).

Could you try to delete the svgAndAwt.svg file and run TestAWT again ?
screen shot 2015-11-21 at 17 35 14

**Comment by anqitong** [comment]
I referenced the problem to #67 : 19_drawing_language. Not this branch 50_add_path

**Comment by starsasumi** [comment]
I think I can make text bold in AWT, but only bold. If it makes sense to all of us, we could also make bold text when width > 1.

**Comment by anqitong** [comment]
yes, I think it is a better soltion to choose a level. Is 1 enough? Maybe starting from 2 or 3? What do you think? @Thomas-dot-G

**Comment by starsasumi** [comment]
@anqitong Ah, got it. It happens because the two branches are all based on master. After we merge these two branch, all should be fine.

**Comment by Thomas-dot-G** [comment]
I'm not sure what will be fined, but we should keep the stroke in xml for pencil. If needed, we could just disable it in awt. (put something by default or just not use it)

**Comment by starsasumi** [comment]
Yeah, I am agree with you. we should keep the class structure for Text. I will try to make text bold in awt.

@Thomas-dot-G Also, do you think we could remove arc in LineType? After studying it all day, I found no easy way to implement arc in AWT. And parameters that an arc takes in xml also can't fit our `PartPath`. 

Paras for an arc in xml: 

```
A rx ry x-axis-rotation large-arc-flag sweep-flag x y
```

As you can see, it needs a point(rx, ry), a `x-axis-rotation`, two flag that could be in a point, and another point(x,y). But in our `PartPath`, we only use points as paras. I understand we can use a point to store `x-axis-rotation`. 

On the other hand, drawing an arc in java is like this:

```
public abstract void drawArc(
                         int x,
                         int y,
                         int width,
                         int height,
                         int startAngle,
                         int arcAngle)
```

My math is just too bad to implement this ...

**Comment by Thomas-dot-G** [comment]
Ok, I'll try to see that. Because I guess we can just translate the 3 points into a circle arc (I guess their is only one way to do that), so I can try to translate into both method. I still need to see that for xml, and I'll keep you inform for awt

**Comment by Thomas-dot-G** [comment]
Ok, we will just no use the Arc^^
I will still keep it, but don't allow it to be used. The same shapes can still be added differently

**Comment by starsasumi** [comment]
Hi, now I make text bold when it's width is larger than 3, and keep arc draw nothing in awt. Would you two see whether it is ok now? ;)

**Comment by Thomas-dot-G** [comment]
capture du 2015-11-22 15 58 20

There is still a little glitch for the line and the position of the circle.
Could you for the test reduce the line width of the svg, so it looks more alike.

And then, could you add that "test" class in the current main ? Maybe create a small test where you just check that the creation of some elements don't change or that passing between awt and svg with the same parameters does not throw exception.

Then also could you add some comments in that class (creation of window...)

Otherwise it looks great and you managed to remove the filled polyline

**Comment by starsasumi** [comment]
Bugs got fixed and I separate JUnit test and the drawing for awt (We could delete this later). All should be fine now. 

Can I merge?

File: src/model/presentation/awt/AWTFrame.java [hunk 1]
```diff
@@ -24,10 +23,32 @@
...
+                g2d = (Graphics2D) g;
+                for (Shape shape : shapes) {
+                    getShapeState(shape).getDrawing();
+                }
+            }
+
+            @Override
+            public Dimension getPreferredSize() {
+                Dimension preferredSize = super.getPreferredSize();
+                // Let's make sure that we have at least our little square size.
```
Comment (Thomas-dot-G) [inline src/model/presentation/awt/AWTFrame.java]: Not sure what to draw from this comment. What do you mean ?

File: src/model/presentation/awt/AWTFrame.java [hunk 2]
```diff
@@ -121,4 +134,4 @@ public Presentation clone() {
     public void insert(Presentation p) {
         //TODO
     }
-}
+}
```
Comment (Thomas-dot-G) [inline src/model/presentation/awt/AWTFrame.java]: Add an empty line here

File: src/model/presentation/awt/AWTPath.java [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    private Path path;
+    private StateDelegate delegate;
+
+    public AWTPath(Path path, StateDelegate delegate) {
+        this.delegate = delegate;
+        this.path = path;
+    }
+
+    @Override
+    public Object getDrawing() {
```
Comment (Thomas-dot-G) [inline src/model/presentation/awt/AWTPath.java]: Would you mind commenting that code please a little more

**Comment by Thomas-dot-G** [comment]
Except for some minor changes and some conflicts that will be ready to merge soon !
In the read me we will need to focus on the fact that even if there is several class where the user can implement code, anything can be done in one only.

**Comment by starsasumi** [comment]
Add lots of javadocs and comments handle conflicts with other merged PR. Is it good to go?

**Comment by Thomas-dot-G** [comment]
Ok for me

**Comment by anqitong** [comment]
Good for merge

**Pull request merged**